### PR TITLE
New: ecmaFeatures.impliedStrict (fixes: #227)

### PR DESCRIPTION
**Pull request by nre**
- New feature `ecmaFeatures.impliedStrict`.
- README.
- Two tests.
- Modified tests/lib/ecma-features.js:
  - `ecmaFeatures[feature] = true` throws for `impliedStrict`.

### CODE REVIEW

File: tests/lib/ecma-features.js [hunk 1]
```diff
@@ -75,17 +75,18 @@ describe("ecmaFeatures", function() {
         // Uncomment and fill in filename to focus on a single file
         // var filename = "jsx/invalid-matching-placeholder-in-closing-tag";
         var feature = path.dirname(filename),
+            permissiveValue = feature !== "impliedStrict",
```
Comment (nzakas) [inline tests/lib/ecma-features.js]: Why this change?

File: tests/fixtures/ecma-features/impliedStrict/inner-strict-violation.src.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+function inner() {
+    var private;
```
Comment (nzakas) [inline tests/fixtures/ecma-features/impliedStrict/inner-strict-violation.src.js]: This should throw a parsing error, correct? If so, the result file should have an error object, not an AST.

File: tests/lib/ecma-features.js (hunk 1 above)
Comment (nzakas) [inline tests/lib/ecma-features.js]: Ooh, I think you're doing this because the implied strict tests are failing. Please revert this change and fix the tests.

**Comment by nre** [comment]
The reason I changed ecma-features.js is that I don't know any errors that are silenced by strict mode. In other words, what code will make a 'non-strict' parser throw, but not a 'strict' parser? So for `impliedStrict` I swapped the 'correct/error' roles of the true/false feature settings.

The other ecmaFeatures,

- `jsx`
- `globalReturn`
- `experimentalObjectRestSpread`

all solve parser errors when they are `true`. But `impliedStrict` finds more errors when it is `true`.

The `impliedStrict` tests generate the ASTs in the result files when `impliedStrict: false`, and throw when `impliedStrict: true`. This is the opposite of the other ecmaFeatures tests, but I don't think that means they are failing.

**Comment by nzakas** [comment]
Ah, I see, thanks for explaining. We should split out that case in a more obvious way, then. Maybe create an array called `failOnEnable` and put `impliedStrict` in it?

**Comment by nre** [comment]
I moved the condition inside `shouldThrowInTestsWhenEnabled(feature)`.

**Comment by nzakas** [comment]
Fair enough, thanks!
